### ISSUE TITLE
bug(tune): GPU trainer Adam/AdamW/SGD-momentum optimizer step submits an empty command buffer and reports success (FP-165)

### ISSUE THREAD

**Issue by ohdearquant** [ohdearquant/lattice#797]
> _Issue authored by Claude (Anthropic agent) on behalf of @ohdearquant._

## Summary

`GpuOptimizer::dispatch_optimizer_update` (crates/tune/src/train/gpu/optimizers.rs:121-146) never encodes a compute pass. It creates the uniform buffer, fetches the pipeline's bind-group layout, then submits an **empty** command encoder and returns `Ok(())`:

```rust
// TODO(FP-165): submits no actual compute — needs access to weight buffers from GpuNetwork
let encoder = device.create_command_encoder(&wgpu::CommandEncoderDescriptor::default());
queue.submit(std::iter::once(encoder.finish()));
Ok(())
```

All three shader-based optimizers route through it (`update`, :156-161): **Adam, AdamW, SGD-momentum**. After the no-op "update", the timestep still increments (:167-169) and the trainer proceeds to weight sync — so GPU training with any of these optimizers silently performs zero weight updates while reporting success at every step.

The path is live, not dead code: `GpuTrainer::update_weights` (crates/tune/src/train/gpu/mod.rs) calls `GpuOptimizer::update` in the training loop.

## Adjacent silent behavior (same match arm)

`Optimizer::RMSprop` silently falls back to plain SGD (:163, `// Fallback`) with no warning — a user selecting RMSprop gets a different algorithm without any indication. Should fail loudly or log prominently as part of the same fix.

## Expected fix (fail-loud first)

Per the tech-debt design review: until the weight/gradient buffers are actually wired into the bind group, these three optimizer arms must return a `TuneError::Training` error ("GPU {optimizer} update not implemented: optimizer shader dispatch has no buffer bindings") instead of succeeding as a no-op. Wiring the real dispatch is the follow-up; the immediate defect is success-without-update.

- The fail-loud fix must be mutation-sensitive: a test that selects Adam on the GPU trainer and asserts the error fires (and fails if the empty-submit `Ok(())` is restored).
- Do not remove the `FP-165` marker without closing the behavior; replace it with a reference to this issue.

## Classification

- Live correctness bug in published `lattice-tune` (crates.io), not stale debt.
- Severity: high — silent training no-op, plausibly interpreted as "model converged" or "training is just slow".

Found by the 2026-07-10 tech-debt audit (dead-code cluster review); verified at source before filing.

**Comment by ohdearquant** [comment]
> _Comment authored by Claude (Anthropic agent) on behalf of @ohdearquant._

Scope update from the fix round: the plain-SGD arm (`update_sgd`, optimizers.rs:113-137) has the same defect, and worse — it does not read real gradients at all (`let grad_scale = 0.01; // Placeholder gradient magnitude`, i.e. it subtracts a constant), then discards the computed vectors because `GpuNetwork` exposes no mutable write-back (`let _ = (weights, biases);`). So all five optimizer arms (Adam, AdamW, SGD-momentum, SGD, RMSprop-via-SGD) currently report success while performing zero (or meaningless) updates.

The fail-loud fix in #798 is being extended to cover the SGD arm as well: `GpuOptimizer::update` will return an error for every optimizer until the real buffer wiring lands. That makes the GPU trainer honestly unusable rather than silently unusable, which is its actual state. The follow-up implementation issue (buffer bindings + mutable weight access + real gradient plumbing) covers all arms.
